include/rocksdb/db.h

```diff
@@ -602,6 +602,24 @@ class DB {
     return Status::NotSupported("GetEntity not supported");
   }
 
+  // Returns logically grouped wide-column entities per column family (a.k.a.
+  // attribute groups) for a single key. PinnableAttributeGroups is a vector of
+  // PinnableAttributeGroup. Each PinnableAttributeGroup will have
+  // ColumnFamilyHandle*, Status and PinnableWideColumns as output.
+  virtual void GetEntity(const ReadOptions& /* options */,
+                         const Slice& /* key */,
+                         PinnableAttributeGroups& result) {
```

Comment:
Alternatively, we can take `ColumnFamilyHandle**`, `num_column_families` and construct result in `PinnableAttributeGroups` and return.